### PR TITLE
Detect priority modules

### PR DESCRIPTION
**Pull request by tanny411**
Added Script to detect priority modules. It runs through the feature list and returns 'what percentile each number is' in their respective wikis as a 'score' per feature. The scores are then saved as a `csv` file. 
A additional function is present in the script, given feature list and weights, it returns a `score per module`.

### CODE REVIEW

File: get_distribution.py [hunk 1]
```diff
@@ -0,0 +1,249 @@
...
+            cur.fetchall(),
+            columns=cols,
+        ).applymap(encode_if_necessary)
+    close_conn(conn)
+
+    df["c_frq"] = (df["edits"] / df["editors"]).replace(np.inf, 0)
+    time_range = (df["last_edit"] - df["first_edit"]).apply(
+        lambda x: x.days + x.seconds / (24 * 60 * 60)
+    )
+    df["t_frq"] = (df["edits"] / time_range).replace(np.inf, 0)
```
Comment (Retailed) [inline get_distribution.py]: I think `t_frq` and `c_frq` upper is not really a good naming. Frq is easily transcribed, but `t` and `c` really catch the eye.

File: get_distribution.py [hunk 2]
```diff
@@ -0,0 +1,249 @@
...
+
+def get_multipliers(df, threshold=0.87):
+    """
+    Alters distribution of each column in df such that the heuristic values in
+    'limits_num' dictionary is the 'threshold' percentile or less. This ensures
+    higher values of certain features get more priority, and the 99.9% of values
+    (which are very low) are not prioritized too much. This does NOT change original
+    values but simply removes lower values when calculating percentiles.
+
+    :param value: The value whose percentile is required
```
Comment (Retailed) [inline get_distribution.py]: Parameters and their description don't match

File: get_distribution.py [hunk 3]
```diff
@@ -0,0 +1,249 @@
...
+    :param d: A pandas.Series or Numpy array which represents the distribution
+    :return: dictionary of heuristic limits, dictionary of multipliers
+    """
+
+    # List of determined heuristics
+    # So far for options between norm and actual numbers, norms are used
+    limits_num = {
+        "c_frq": 200,
```
Comment (Retailed) [inline get_distribution.py]: Naming for `c_frq` and `t_frq` again

File: get_distribution.py [hunk 4]
```diff
@@ -0,0 +1,249 @@
...
+        p = limits_perc[k]
+        d = df[df[k] >= p * v][k]
+        df[k + "_score"] = df[k].apply(lambda x: which_percentile(x, d))
+        del df[k]
+    return df
+
+
+def get_score(
+    feature_names=[
+        "c_frq_score",
```
Comment (Retailed) [inline get_distribution.py]: Last one naming thing!

**Comment by tanny411** [comment]
Hi, I'v updated some docstring and feature names, let me know if anything else is unclear.
I've also fixed data_modules bash script.

File: get_distribution.py (hunk 2 above)
Comment (Retailed) [reply get_distribution.py]: @tanny411 I believe this one is still missing, as the names of parameters in the comment are different.

**Review by Retailed — COMMENTED** [review]
> Hi, I'v updated some docstring and feature names, let me know if anything else is unclear.
> I've also fixed data_modules bash script.

Seems like there's still a bit of the problem with the docstring - but all the other things are clear. Thanks a lot for fixing bash script!

File: get_distribution.py (hunk 2 above)
Comment (tanny411) [reply get_distribution.py]: Not sure why github wouldn't show me the comments earlier, so I had fixed scripts on my judgemnet. Incorporated your suggestion now.
Thanks!

**Pull request merged**